core/events/events.js

```diff
@@ -14,201 +14,286 @@
  * Events fired as a result of actions in Blockly's editor.
  * @namespace Blockly.Events
  */
-goog.provide('Blockly.Events');
-
-goog.require('Blockly.registry');
-goog.require('Blockly.utils');
-
-goog.requireType('Blockly.Block');
-goog.requireType('Blockly.Events.Abstract');
-goog.requireType('Blockly.Workspace');
+goog.module('Blockly.Events');
+goog.module.declareLegacyNamespace();
+
+/* eslint-disable-next-line no-unused-vars */
+const Abstract = goog.requireType('Blockly.Events.Abstract');
+/* eslint-disable-next-line no-unused-vars */
+const Block = goog.requireType('Blockly.Block');
+/* eslint-disable-next-line no-unused-vars */
+const BlockCreate = goog.requireType('Blockly.Events.BlockCreate');
+/* eslint-disable-next-line no-unused-vars */
+const BlockMove = goog.requireType('Blockly.Events.BlockMove');
+/* eslint-disable-next-line no-unused-vars */
+const CommentCreate = goog.requireType('Blockly.Events.CommentCreate');
+/* eslint-disable-next-line no-unused-vars */
+const CommentMove = goog.requireType('Blockly.Events.CommentMove');
+/* eslint-disable-next-line no-unused-vars */
+const Workspace = goog.requireType('Blockly.Workspace');
+const deprecation = goog.require('Blockly.utils.deprecation');
+const registry = goog.require('Blockly.registry');
+const utils = goog.require('Blockly.utils');
 
 
 /**
  * Group ID for new events.  Grouped events are indivisible.
  * @type {string}
- * @private
  */
-Blockly.Events.group_ = '';
+let group = '';
 
 /**
  * Sets whether the next event should be added to the undo stack.
  * @type {boolean}
  */
-Blockly.Events.recordUndo = true;
+let recordUndo = true;
+/** @deprecated September 2021 */
+exports.recordUndo = recordUndo;
+
+/**
+ * Sets whether events should be added to the undo stack.
+ * @param {boolean} newValue True if events should be added to the undo stack.
+ */
+const setRecordUndo = function(newValue) {
+  recordUndo = newValue;
+};
+exports.setRecordUndo = setRecordUndo;
+
+/**
+ * Returns whether or not events will be added to the undo stack.
+ * @returns {boolean} True if events will be added to the undo stack.
+ */
+const getRecordUndo = function() {
+  return recordUndo;
+};
+exports.getRecordUndo = getRecordUndo;
+
+Object.defineProperties(exports, {
+  recordUndo: {
+    get: function() {
+      deprecation.warn(
+        'Blockly.Events.recordUndo', 'September 2021', 'September 2022',
+        'Blockly.Events.getRecordUndo()');
+      return getRecordUndo();
+    },
+    set: function(record) {
+      deprecation.warn(
+        'Blockly.Events.recordUndo', 'September 2021', 'September 2022',
+        'Blockly.Events.setRecordUndo()');
+      setRecordUndo(record);
+    },
+  }
+ });
 
 /**
  * Allow change events to be created and fired.
  * @type {number}
- * @private
  */
-Blockly.Events.disabled_ = 0;
+let disabled = 0;
+/** @private */
+exports.disabled_ = disabled;
```

Comment:
This line appears to be clobbered by the setter below.  Delete?

Reply:
Closure isn't able to see properties that get added by Object.defineProperties, so this is here solely to inform the compiler that a .disabled_ property exists.